tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/Package/PackageCheckTool.cs

```diff
@@ -120,68 +120,124 @@ private async Task<CLICheckResponse> RunAllChecks(string packagePath, bool fixCh
             var results = new List<CLICheckResponse>();
             var overallSuccess = true;
             var failedChecks = new List<string>();
+            var notImplementedChecks = new List<string>();
 
             // Run dependency check
-            var dependencyCheckResult = await languageChecks.AnalyzeDependenciesAsync(packagePath, fixCheckErrors, ct);
+            var languageChecks = await ResolveLanguageChecks(packagePath, ct);
+            if (languageChecks == null)
+            {
+                return CreateUnsupportedLanguageResponse(packagePath);
+            }
+
+            var dependencyCheckResult = await languageChecks.AnalyzeDependencies(packagePath, fixCheckErrors, ct);
             results.Add(dependencyCheckResult);
             if (dependencyCheckResult.ExitCode != 0)
             {
-                overallSuccess = false;
-                failedChecks.Add("Dependency");
+                if (IsNotImplemented(dependencyCheckResult))
+                {
+                    notImplementedChecks.Add("Dependency");
+                }
+                else
+                {
+                    overallSuccess = false;
+                    failedChecks.Add("Dependency");
+                }
             }
 
             // Run changelog validation
-            var changelogValidationResult = await languageChecks.ValidateChangelogAsync(packagePath, fixCheckErrors, ct);
+            var changelogValidationResult = await languageChecks.ValidateChangelog(packagePath, fixCheckErrors, ct);
             results.Add(changelogValidationResult);
             if (changelogValidationResult.ExitCode != 0)
             {
-                overallSuccess = false;
-                failedChecks.Add("Changelog");
+                if (IsNotImplemented(changelogValidationResult))
+                {
+                    notImplementedChecks.Add("Changelog");
+                }
+                else
+                {
+                    overallSuccess = false;
+                    failedChecks.Add("Changelog");
+                }
             }
 
             // Run README validation
-            var readmeValidationResult = await languageChecks.ValidateReadmeAsync(packagePath, fixCheckErrors, ct);
+            var readmeValidationResult = await languageChecks.ValidateReadme(packagePath, fixCheckErrors, ct);
             results.Add(readmeValidationResult);
             if (readmeValidationResult.ExitCode != 0)
             {
-                overallSuccess = false;
-                failedChecks.Add("README");
+                if (IsNotImplemented(readmeValidationResult))
```

Comment:
Maybe we should have a different way of saying "not implemented" instead of checking the next steps message? Like we could

- have a special static CLICheckResponse.NotImplemented that we check for equality against and non-implemented versions return CLICheckResponse.NotImplemented
- just have the non-implemented methods throw NotImplementedException and catch